pkg/local_object_storage/metabase/containers.go

```diff
@@ -104,3 +107,113 @@ func changeContainerSize(tx *bbolt.Tx, id cid.ID, delta uint64, increase bool) e
+
+// DeleteContainer removes any information that the metabase has
+// associated with the provided container (its objects).
+//
+// Note: does not clean graveyard.
+func (db *DB) DeleteContainer(cID cid.ID) error {
+	db.modeMtx.RLock()
+	defer db.modeMtx.RUnlock()
+
+	if db.mode.NoMetabase() {
+		return ErrDegradedMode
+	} else if db.mode.ReadOnly() {
+		return ErrReadOnlyMode
+	}
+
+	cIDRaw := make([]byte, cidSize)
+	cID.Encode(cIDRaw)
+	buff := make([]byte, addressKeySize)
+
+	return db.boltDB.Update(func(tx *bbolt.Tx) error {
+		// Estimations
+		bktEstimations := tx.Bucket(containerVolumeBucketName)
+		err := bktEstimations.Delete(cIDRaw)
+		if err != nil {
+			return fmt.Errorf("estimations bucket cleanup: %w", err)
+		}
+
+		// Locked objects
+		bktLocked := tx.Bucket(bucketNameLocked)
+		err = bktLocked.DeleteBucket(cIDRaw)
+		if err != nil && !errors.Is(err, bbolt.ErrBucketNotFound) {
+			return fmt.Errorf("locked bucket cleanup: %w", err)
+		}
+
+		// Regular objects
+		err = tx.DeleteBucket(primaryBucketName(cID, buff))
+		if err != nil && !errors.Is(err, bbolt.ErrBucketNotFound) {
+			return fmt.Errorf("regular bucket cleanup: %w", err)
+		}
+
+		// Lock objects
+		err = tx.DeleteBucket(bucketNameLockers(cID, buff))
+		if err != nil && !errors.Is(err, bbolt.ErrBucketNotFound) {
+			return fmt.Errorf("lockers bucket cleanup: %w", err)
+		}
+
+		// SG objects
+		err = tx.DeleteBucket(storageGroupBucketName(cID, buff))
+		if err != nil && !errors.Is(err, bbolt.ErrBucketNotFound) {
+			return fmt.Errorf("SG bucket cleanup: %w", err)
+		}
+
+		// TS objects
+		err = tx.DeleteBucket(tombstoneBucketName(cID, buff))
+		if err != nil && !errors.Is(err, bbolt.ErrBucketNotFound) {
+			return fmt.Errorf("TS bucket cleanup: %w", err)
+		}
+
+		// Small objects
+		err = tx.DeleteBucket(smallBucketName(cID, buff))
+		if err != nil && !errors.Is(err, bbolt.ErrBucketNotFound) {
+			return fmt.Errorf("small bucket cleanup: %w", err)
+		}
+
+		// Root objects
+		err = tx.DeleteBucket(rootBucketName(cID, buff))
+		if err != nil && !errors.Is(err, bbolt.ErrBucketNotFound) {
+			return fmt.Errorf("root bucket cleanup: %w", err)
+		}
+
+		// indexes
+
+		err = tx.DeleteBucket(ownerBucketName(cID, buff))
+		if err != nil && !errors.Is(err, bbolt.ErrBucketNotFound) {
+			return fmt.Errorf("owner index cleanup: %w", err)
+		}
+
+		err = tx.DeleteBucket(payloadHashBucketName(cID, buff))
+		if err != nil && !errors.Is(err, bbolt.ErrBucketNotFound) {
+			return fmt.Errorf("hash index cleanup: %w", err)
+		}
+
+		err = tx.DeleteBucket(parentBucketName(cID, buff))
+		if err != nil && !errors.Is(err, bbolt.ErrBucketNotFound) {
+			return fmt.Errorf("parent index cleanup: %w", err)
+		}
+
+		err = tx.DeleteBucket(splitBucketName(cID, buff))
+		if err != nil && !errors.Is(err, bbolt.ErrBucketNotFound) {
+			return fmt.Errorf("split id index cleanup: %w", err)
+		}
+
+		// Attributes index
+		var keysToDelete [][]byte // see https://github.com/etcd-io/bbolt/issues/146
```

Comment:
How many of those can we expect?

Reply:
any number. any container can be dropped and it may have this or that number